py-doc-to-xml/docx_to_xml/types.py

```diff
@@ -25,6 +25,28 @@ class SemanticDomain:
     description: str
     questions: List[str]
 
-    def is_valid(self):
+    def is_valid(self) -> bool:
         return is_semantic_domain_number(self.number)
-    
\ No newline at end of file
+
+    # # Define the comparison operators so that Python will sort for us
+    # # based only on the domain number
+    def __str__(self) -> str:
```

Comment:
`dataclass`es have built-in support for defining how to sort them to avoid the boilerplate below.

- https://www.pythontutorial.net/python-oop/python-dataclass/
  - Scroll down to **Sort objects**

Reply:
Did you try the steps on the article above to opt-in to `dataclass` ordering?

Reply:
When I implement the suggested method for sorting (which is much cleaner), an exception is thrown because `frozen=True`.  Is the reason for frozen instances so that they can be used efficiently in lists & dicts (& probably `lxml`)?

Reply:
`frozen=True` is the default setting I use for all `dataclass`es as it makes instances immutable, which helps with reasoning about the code and limiting the scope of what can be done with those instances.

Could you share the `Exception` you got?

Reply:
The exception is:
```
Exception has occurred: FrozenInstanceError
cannot assign to field 'sort_index'

File "[/home/grady/projects/sil/sem-dom/py-doc-to-xml/docx_to_xml/types.py]()", line 30, in __post_init__
    self.sort_index = self.number
```
The dataclass, based on the instructions, is now:
```
@dataclass(frozen=True, order=True)
class SemanticDomain:
    """Intermediate representation of a Semantic Domain."""

    number: str
    title: str
    description: str
    questions: List[str]
    sort_index: int = field(init=False, repr=False)

    def __post_init__(self):
        self.sort_index = self.number

    def is_valid(self) -> bool:
        return is_semantic_domain_number(self.number)
```

Reply:
There is a sort gross work around for this here: https://stackoverflow.com/questions/53756788/how-to-set-the-value-of-dataclass-field-in-post-init-when-frozen-true

Let's just drop `frozen=True` from this `dataclass`.

Reply:
OK